### ISSUE TITLE
Expect XY data in geometry build methods by default

### ISSUE THREAD

**Issue by navispatial** [navibyte/geospatial#154]
Now required to specify the coordinate type when building geometry objects:

```dart
Polygon.build([[30, 10, 40, 40, 20, 40, 10, 20, 30, 10]], type: Coords.xy)
```

This information is essential - but it could have a default value `Coords.xy`, so when 2D data is built, the sample would be simplified:

```dart
Polygon.build([[30, 10, 40, 40, 20, 40, 10, 20, 30, 10]])
```

However for `Coords.xyz`, `Coords.xym` and `Coords.xyzm` data this type would still be required.

**Comment by navispatial** [comment]
This is not breaking previous usage, but add a default value to a required parameter.

For example for `Polygon` previous signature:

```dart
factory Polygon.build(
    Iterable<Iterable<double>> rings, {
    required Coords type,
    Iterable<double>? bounds,
  };
```

And the new signature:

```dart
factory Polygon.build(
    Iterable<Iterable<double>> rings, {
    Coords type = Coords.xy,
    Iterable<double>? bounds,
  };
```
After this `type` can be omitted, the default `Coords.xy` is used, an coordinate arrays are expected to contain 2D data (x, y).

**Comment by navispatial** [comment]
[Implemented in version 0.3.2](https://pub.dev/packages/geobase/versions/0.3.2)
